test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -213,6 +213,14 @@ object Assertion {
   final def containsString(element: String): Assertion[String] =
     Assertion.assertion("containsString")(param(element))(_.contains(element))
 
+  final def hasSameElements[A](other: Iterable[A]): Assertion[Iterable[A]] =
+    Assertion.assertionRec[Iterable[A]]("hasSameElements")(param(other)) { (_, actual) =>
+      val actualSeq = actual.toSeq
+      val otherSeq  = other.toSeq
+
+      isEmpty(actualSeq.diff(otherSeq)) && isEmpty(otherSeq.diff(actualSeq))
```

Comment:
Looks great. Can we get a test that the string representation of running the assertion looks like we expect? Then we will be good to go!